V-03 / F-04 выполнены

Feintty opened the pull request.

Feintty committed:
Merge pull request #7 from Feintty/review

Review

Feintty committed:
V-03 Выполнена верстка радиокнопок
V-03 Создана верстка карточки автомобиля
V-03 Выполнена верстка вкладки
F-03 Выполнено получение данных
F-03 Реализована фильтрация данных

Feintty committed:
F-04 Сокращено дублирование кода

konkonskiy commented on need-for-drive/src/components/CarsTab/CarsTab.js:
```diff
@@ -0,0 +1,110 @@
+        cars
+          .filter((el) => {
+            if (el.categoryId) {
+              if (el.categoryId.name) {
```
Чтобы не делать двойную проверку в новых стандартах es есть оператор if (el.categoryId?.name)
Прочитай про него

konkonskiy commented on need-for-drive/src/components/CarsTab/CarsTab.js:
```diff
@@ -0,0 +1,110 @@
+  const [cars, setCars] = useState()
+  const [currentFilterName, setCurrentFilterName] = useState("Все модели")
+
+  const fetchCars = () => {
```
Все что связано с апи выносим в отдельный файл

konkonskiy commented on need-for-drive/src/components/CarsTab/CarsTab.js:
```diff
@@ -0,0 +1,110 @@
+    })
+  }
+
+  const createRadios = () => {
```
Запрашивай категории тож на серваке

konkonskiy commented on need-for-drive/src/components/LocationTab/LocationTab.js:
```diff
@@ -1,22 +1,222 @@
+  const fetchCitiesLocation = () => {
+    const cityNames = citiesList.map((el) => `&location=${el.name}`).join("")
+    fetch(
+      `http://mapquestapi.com/geocoding/v1/batch?key=${process.env.REACT_APP_MAPQUEST_KEY}${cityNames}`,
```
адрес в переменную

konkonskiy commented on need-for-drive/src/components/LocationTab/LocationTab.js:
```diff
@@ -1,22 +1,222 @@
+  const fetchCities = () => {
+    fetch(`${process.env.REACT_APP_API_URL}/api/db/city`, {
+      method: "GET",
+      headers: {
```
У тебя много одинаковых хедеров. Повод вынести

konkonskiy commented on need-for-drive/src/components/LocationTab/LocationTab.js:
```diff
@@ -1,22 +1,222 @@
+
+  useEffect(() => {
+    cityAndPointToOrder(currentCity, currentPoint)
+  }, [currentCity, currentPoint])
```
Подумай как сократить кол-во функций

konkonskiy commented on need-for-drive/src/components/RadioButton/RadioButton.js:
```diff
@@ -0,0 +1,29 @@
+  }
+
+  return (
+    <div className={`radio-button ${isChecked ? " checked" : ""}`}>
```
Чем тебя инпут тайп радио не устроил?

Feintty committed:
F-04 Сокращены двойные if
F-04 Вынесены API запросы
F-04 Категории подтягиваются с сервера
F-04 Переменные url вынесены в окружение
F-04 Переписаны радио баттоны

Feintty committed:
F-04 Установка цены авто

Feintty committed:
F-04 Пофикшена подсветка вкладок

Feintty committed:
F-04 Фикс маркеров
F-04 Названия пунктов заменены на адреса
